VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -2278,44 +2347,109 @@ Function Install-VCFCertificate {
 
     $pass = Get-Password -User $user -Password $pass
 
-    if (!($NoConfirmation.IsPresent) -and ($vsanDataMigrationMode -eq "EnsureAccessibility")) {
-        $warningMessage = "Please ensure sufficient backups of the cluster exists.  Please ensure the ESXi`n"
-        $warningMessage += " hosts activities are minimumized during certificate replacement process. `n"
-        $warningMessage += "Please enter yes to confirm: "
-        $proceed = Read-Host $warningMessage
-        if (($proceed -match "no") -or ($proceed -match "yes")) {
-            if ($proceed -match "no") {
-                return "Stopping script execution. (confirmation is $proceed)."
+    if ($PSBoundParameters.ContainsKey("esxi")){
+        # VCF version checks
+        if (Test-VCFConnection -server $server) {
+			if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+            } else {
+                Throw "Unable to return vCenter Server details: PRE_VALIDATION_FAILED"
             }
         } else {
-            return "None of the options is selected. Default is 'No', hence stopping script execution."
+            Throw "Unable to obtain access token from SDDC Manager ($server), check credentials: PRE_VALIDATION_FAILED"
         }
-    }
-
-    if ($PSBoundParameters.ContainsKey("esxi")){
-        if (!$PSBoundParameters.ContainsKey("cluster") -and !$PSBoundParameters.ContainsKey("esxiFqdn")) {
-            Write-Error "Please provide either -cluster or -esxiFqdn paramater."
-        } elseif ($PSBoundParameters.ContainsKey("cluster") -and $PSBoundParameters.ContainsKey("esxiFqdn")) {
-            Write-Error "Only one of -esxiFqdn or -cluster parameter can be provided at a time."
-        } elseif ($PSBoundParameters.ContainsKey("cluster")) {
-            if ($vsanDataMigrationMode.IsPresent) {
-                if ($migratePowerOffVMs.IsPresent) {
-                    Install-EsxiCertificate -server $server -user $user -pass $pass -domain $domain -cluster $cluster -certificateDirectory $certificateDirectory -certificateFileExt $certificateFileExt -vsanDataMigrationMode $vsanDataMigrationMode -migratePowerOffVMs
+        $version = Get-VCFManager -version
+        $vcfVersion = $version.Split('.')[0]+"."+$version.Split('.')[1]
+
+        if ($vcfVersion -eq "5.2") {
```

Comment:
-ge ?

Reply:
addressed in commit e6a7844965249faebca841b4da521bef3b7b40ba